src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,10 +1753,13 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
```

Comment:
TokenCacheItem is created from AuthenticationResult. what do you mean by removing all of them?
Let's say if i found a regular refresh token or mrrt, and i try to get AT with it. But refresh token request fail(expired), then removeCacheItem will try to remove the following keys: 1) the one used to find RT or MRRT 2) key with user id 3) key with displayable

For family client id refresh token, if the one we use to get AT expired, we won't remove it. You can only update your own entry (with matched client id, authority, resource).